tests/test_events.py

```diff
@@ -327,3 +332,49 @@ def test_event_is_upcoming(event, mocker):
 
     with freeze_time(tomorrow_morning):
         assert not test_event.is_upcoming
+
+
+def test_delete_button_shows(event, admin_client, django_user_model, mocker):
+    # create 1 event with a fake api_source and use requests_mock to come up with 2 different responses (an ok and a 404)
+    e = event.build()
+    event_template = reverse('lametro:events', args=[e.slug])
+    api_source = f'http://webapi.legistar.com/v1/metro/events/{e.slug}'
+    
+    mock_source = mocker.MagicMock()
+    mock_source.url = api_source
+    mock_api_source = mocker.patch('lametro.models.LAMetroEvent.api_source', new_callable=mocker.PropertyMock, return_value=mock_source)
+
+    with requests_mock.Mocker() as m: 
+        cal_matcher = re.compile('https://metro.legistar.com/calendar.aspx')
+        m.get(cal_matcher, status_code=200)
+
+        source_matcher = re.compile(api_source)
+        success = m.get(source_matcher, status_code=200)
+        success_response = admin_client.get(event_template)
+
+        failure = m.get(source_matcher, status_code=404)
+        failure_response = admin_client.get(event_template)
+
+        text = 'This event does not exist in Legistar. It may have been deleted from Legistar due to being a duplicate. To delete this event, click the button below.'
+        assert text not in success_response.content.decode('utf-8')
+        assert text in failure_response.content.decode('utf-8')
+
+
+@pytest.mark.django_db
+def test_delete_event(event, client, admin_client):
+    e = event.build()
+    e.save()
+    event_in_db = LAMetroEvent.objects.filter(id=e.id)
+    assert event_in_db.exists()
+
+    delete_event = reverse('delete_event', args=[e.slug])
+    admin_redirect_url = reverse('lametro:event')
+
+    user_response = client.get(delete_event)
+    assert user_response.url != admin_redirect_url
+
+    admin_response = admin_client.get(delete_event)
+    assert admin_response.url == admin_redirect_url
```

Comment:
What if you also checked the status code of these responses? I'd expect the unauthenticated request to return a 403, and the authenticated request to return a 200 (or a 301 or 302 redirect).

Reply:
I did check for this, and they both result in a 302 redirect. This is the url that the unauthenticated request attempted to redirect to when I ran my tests: `/accounts/login/?next=/delete-event/8159e708-f892-49cf-84b3-175dad36a7f5/`

I did briefly look in the [django docs](https://docs.djangoproject.com/en/3.1/topics/auth/default/#django.contrib.auth.decorators.login_required) to see if there was a way to avoid that redirect but didn't come across anything that would allow me to throw a 403 error, I only found a way to change the `REDIRECT_URL`. Let me know what you think as far as whether I should make any other changes